python/pyarrow/tests/test_hdfs.py

```diff
@@ -321,7 +321,8 @@ def test_read_multiple_parquet_files_with_uri(self):
 
         expected = self._write_multiple_hdfs_pq_files(tmpdir)
         path = _get_hdfs_uri(tmpdir)
-        result = pq.read_table(path)
+        # TODO for URI it should not be needed to pass this argument
+        result = pq.read_table(path, use_legacy_dataset=True)
```

Comment:
This _should_ not be necessary, I would think, since reading from a URI should work with the new datasets implementation as well. 

The error is `pyarrow.lib.ArrowInvalid: Path '/tmp/pyarrow-test-838/multi-parquet-uri-48569714efc74397816722c9c6723191/0.parquet' is not relative to '/user/root'`

Reply:
Perhaps create a jira?

Reply:
Yes, was planning to -> https://issues.apache.org/jira/browse/ARROW-10264